OpenEdXMobile/src/main/java/org/edx/mobile/wraper/InAppPurchasesDialog.kt

```diff
@@ -0,0 +1,241 @@
+package org.edx.mobile.wraper
+
+import android.content.DialogInterface
+import androidx.annotation.StringRes
+import androidx.fragment.app.DialogFragment
+import androidx.fragment.app.Fragment
+import org.edx.mobile.R
+import org.edx.mobile.core.IEdxEnvironment
+import org.edx.mobile.exception.ErrorMessage
+import org.edx.mobile.http.HttpStatus
+import org.edx.mobile.module.analytics.Analytics
+import org.edx.mobile.module.analytics.InAppPurchasesAnalytics
+import org.edx.mobile.util.AppConstants
+import org.edx.mobile.util.ResourceUtil
+import org.edx.mobile.util.TextUtils
+import org.edx.mobile.view.dialog.AlertDialogFragment
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class InAppPurchasesDialog @Inject constructor(
+    var environment: IEdxEnvironment,
+    var iapAnalytics: InAppPurchasesAnalytics
+) {
+    /**
```

Comment:
Instead of handling exceptions in multiple classes, we can handle all the exceptions before calling the dialogs. Do add doc for this.
```
fun handleIAPException(
        fragment: Fragment,
        errorMessage: ErrorMessage,
        retryListener: DialogInterface.OnClickListener?,
        cancelListener: DialogInterface.OnClickListener? = null
    ) {
        errorMessage.throwable as InAppPurchasesException
        when (errorMessage.throwable.httpErrorCode) {
            HttpStatus.UNAUTHORIZED -> {
                environment.router?.forceLogout(
                    fragment.requireContext(),
                    environment.analyticsRegistry,
                    environment.notificationDelegate
                )
                return
            }
            else -> {
                cancelListener?.let {
                    showPostUpgradeErrorDialog(
                        context = fragment,
                        errorMessage = errorMessage,
                        retryListener = retryListener,
                        cancelListener = cancelListener
                    )
                } ?: showUpgradeErrorDialog(
                    context = fragment,
                    errorMessage = errorMessage,
                    retryListener = retryListener
                )
            }
        }
    }